src/compiler/rust/canister_methods/src/upload_file_chunk/mod.rs

```diff
@@ -50,6 +46,13 @@ pub fn get_upload_file_chunk() -> proc_macro2::TokenStream {
             }
         }
 
+        pub fn is_authenticated() -> Result<(), String> {
+            if ic_cdk::api::is_controller(&ic_cdk::api::caller()) {
+                return Ok(());
+            }
+            return Err("Access denied: Must be a controller to handle files!".to_string());
```

Comment:
Can we make this generic? Just `Not Authorized: must be a controller to call this method`